pkg/http_endpoint/v1/router.go

```diff
@@ -56,26 +60,645 @@ func authRouter(clients *common.ClientContainer,
+	err := clients.Grafana.DoLogon()
+        if err != nil {
+                switch err.(type) {
+                // common.InvalidUsernamePassword means, that user provided invalid
+                // username/password. We have to return 401 error then
+                case common.InvalidUsernamePassword:
+                        common.WriteErrorToResponse(w, http.StatusUnauthorized,
+                                "Unauthorized", err.Error())
+                        return
+                // If any other error happened -> return 500 error
+                default:
+                        log.Logger.Error(err)
+                        common.WriteErrorToResponse(w, http.StatusInternalServerError,
+                                http.StatusText(http.StatusInternalServerError),
+                                "Internal server error occured")
+                        return
+                }       
+        }
+	datasourceID := chi.URLParam(r, "datasourceID")
+	ID, _ := strconv.Atoi(datasourceID)
+	err = clients.Grafana.DeleteDataSource(ID)
+	if err != nil {
+		log.Logger.Error(err)
+		common.WriteErrorToResponse(w, http.StatusInternalServerError,
+			http.StatusText(http.StatusInternalServerError),
+			"Internal server error occured")
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	w.Write([]byte(fmt.Sprintf("Sucessfully deleted datasource by ID: %v", datasourceID)))
+		}
+}
+
+//CreateDatasource method is to create  datasource
+func CreateDatasource(clients *common.ClientContainer) http.HandlerFunc {
+		return func(w http.ResponseWriter, r *http.Request) {
+	err := clients.Grafana.DoLogon()
+        if err != nil {
+                switch err.(type) {
+                // common.InvalidUsernamePassword means, that user provided invalid
+                // username/password. We have to return 401 error then
+                case common.InvalidUsernamePassword:
+                        common.WriteErrorToResponse(w, http.StatusUnauthorized,
+                                "Unauthorized", err.Error())
+                        return  
+                // If any other error happened -> return 500 error
+                default:
+                        log.Logger.Error(err)
+                        common.WriteErrorToResponse(w, http.StatusInternalServerError,
+                                http.StatusText(http.StatusInternalServerError),
+                                "Internal server error occured")
+                        return  
+                }       
+        }
+	r.ParseForm()
+	for params := range r.Form {
+		res := grafanaclient.DataSource{}
+		json.Unmarshal([]byte(params), &res)
+		err := clients.Grafana.CreateDataSource(res)
+		if err != nil {
+			switch err.(type) {
+			// common.DatasourceExists means, that user provided
+			// datasource details which exist already. we return
+			// 409 error
+			case common.DatasourceExists:
+				common.WriteErrorToResponse(w, 409,
```

Comment:
common.WriteErrorToResponse(w, http.StatusConflict, http.StatusText(http.StatusConflict), "This string has to be changed, because it can be not safe to return error to user. It may contain some creds")